hibernate-core/src/test/java/org/hibernate/test/hql/UpdateEntityWithEmbeddedTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.hql;
+
+import javax.persistence.Embeddable;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.dialect.PostgreSQL81Dialect;
+
+import org.hibernate.testing.RequiresDialect;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+/**
+ * @author Andrea Boriero
+ */
+@TestForIssue(jiraKey = "HHH-14251")
+@RequiresDialect(PostgreSQL81Dialect.class)
```

Comment:
Is the issue dialect specific? Saw the code changes in `DotNode` are generic.

Reply:
Different behaviour based on `Dialect#supportsRowValueConstructorSyntax()` , I'm working on a fix for Dialects not supporting it.

Reply:
> Different behaviour based on `Dialect#supportsRowValueConstructorSyntax()` , I'm working on a fix for Dialects not supporting it.

https://github.com/hibernate/hibernate-orm/blob/41161f9fa9e1d760995c462c78cbd19ef8e7b082/hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java#L456
`MySQLDialect#supportsRowValueConstructorSyntax()` return true, but row value constructor is not supported in set clause, only supported in where clause.

Reply:
Thanks @quaff, I'm working on a possible solution.